### PR TITLE
fixing problems with artifactory uploads

### LINKED ISSUE #3068: fix artifactory setup
The broad artifactory moved to https://broadinstitute.jfrog.io/broadinstitute/.

There is a redirect in place which as been working for downloads, but uploads are failing with `401 Unauthorized`.  It seems like updating the url fixes the problem.

As a second issue, our builds try to upload archives for every integration test build, which worked when we only had 1 integration test build, but now that we have multiples we are uploading duplicates which isn't good.  We should fix that, probably by adding either a new environment variable to the travis build, or a final build stage to perform the upload.

### PR DESCRIPTION
**Pull request by lbergelson**
update the artifactory url to point to the new artifactory
update the travis build with another environment variable called UPLOAD which determines if that build should upload a snapshot or not
fixes #3068

### CODE REVIEW

**Comment by droazen** [comment]
:+1: merge when ready

File: .travis.yml [hunk 1]
```diff
@@ -122,9 +122,9 @@ script:
 
 after_success:
 - bash <(curl -s https://codecov.io/bash)
-- if [[ $TRAVIS_BRANCH == master && $TEST_TYPE == integration ]]; then ./gradlew uploadArchives; fi
+- if [[ $TRAVIS_BRANCH == master && $UPLOAD == true ]]; then ./gradlew uploadArchives; fi
 # if this is actually a commit to master and not a pull request build into master, then publish master-snapshot
-- if [[ $TRAVIS_BRANCH == master && $TRAVIS_PULL_REQUEST == false && $TEST_TYPE == integration ]]; then 
+- if [[ $TRAVIS_BRANCH == master && $TRAVIS_PULL_REQUEST == false && UPLOAD == integration ]]; then
```
Comment (droazen) [inline .travis.yml]: Wait a sec, when would `UPLOAD == integration`?

Comment (lbergelson) [inline .travis.yml]: doh

File: .travis.yml [hunk 2]
```diff
@@ -122,9 +122,9 @@ script:
 
 after_success:
 - bash <(curl -s https://codecov.io/bash)
-- if [[ $TRAVIS_BRANCH == master && $TEST_TYPE == integration ]]; then ./gradlew uploadArchives; fi
+- if [[ $TRAVIS_BRANCH == master && $UPLOAD == true ]]; then ./gradlew uploadArchives; fi
 # if this is actually a commit to master and not a pull request build into master, then publish master-snapshot
-- if [[ $TRAVIS_BRANCH == master && $TRAVIS_PULL_REQUEST == false && $TEST_TYPE == integration ]]; then 
+- if [[ $TRAVIS_BRANCH == master && $TRAVIS_PULL_REQUEST == false && UPLOAD == true ]]; then
```
Comment (droazen) [inline .travis.yml]: `UPLOAD == true` or `$UPLOAD == true` ?

**Pull request merged**